pkg/consoleutil/detach.go

```diff
@@ -0,0 +1,72 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package consoleutil
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"syscall"
+
+	"github.com/moby/term"
+)
+
+var defaultDetachKeys = "ctrl-p,ctrl-q"
+
+type detachableStdin struct {
+	stdin  io.Reader
+	closer func()
+	mu     sync.Mutex
+	closed bool
+}
+
+// NewDetachableStdin returns a new TTY proxy reader
```

Comment:
NewDetachableStdin returns a new `detachableStdin` which uses a TTY proxy reader to read specified detach keys from stdin, in which case closer will be called

Reply:
Hmm, I'm not sure if we want to mention the underlying unexported struct here, maybe the users of this function shouldn't need to know that implementation detail?

Reply:
LGTM, but just one thing was confusing me is : `NewDetachableStdin returns a new TTY proxy reader`. In fact the `NewDetachableStdin` is not returning a TTY proxy reader but it is wrapping a a TTY proxy reader